JsonPatch/PatchExtensions.cs

```diff
@@ -0,0 +1,142 @@
+	{
+		/// <summary>
+		/// Apply patch to object
+		/// </summary>
+		/// <param name="patch">JsonPatch object</param>
+		/// <param name="obj">Object to patch</param>
+		/// <param name="options">Options</param>
+		/// <typeparam name="T">Any object</typeparam>
+		/// <returns>New instance of patched object</returns>
+		public static T ApplyPatch<T>(this Json.Patch.JsonPatch patch, T obj, JsonSerializerOptions options = null)
+		{
+			using var doc = JsonDocument.Parse(JsonSerializer.Serialize(obj, options));
+			var res = patch.Apply(doc.RootElement).Result;
+			var result = JsonSerializer.Deserialize<T>(res.GetRawText(), options);
+			return result;
+		}
+		/// <summary>
+		/// Create JsonPatch from difference between original and modified
+		/// </summary>
+		/// <param name="originalObject">Original object</param>
+		/// <param name="modifiedObject">Modified object</param>
+		/// <param name="options">Json serializer options</param>
+		/// <typeparam name="T">Any object</typeparam>
+		/// <returns>JsonPatch</returns>
+		public static JsonPatch CreatePatch<T>(this T originalObject, T modifiedObject, JsonSerializerOptions options = null)
+        {
+            var original = originalObject.ToJsonDocument(options);
+            var modified = modifiedObject.ToJsonDocument(options);
+
+            return CreatePatch(original, modified);
+        }
+
+		/// <summary>
+		/// Create JsonPatch from difference between original and modified
+		/// </summary>
+		/// <param name="originalObject">Original object</param>
+		/// <param name="modifiedObject">Modified object</param>
+		/// <returns>JsonPatch</returns>
+        public static JsonPatch CreatePatch(this JsonDocument originalObject, JsonDocument modifiedObject)
+        {
+            return CreatePatch(originalObject.RootElement, modifiedObject.RootElement);
+        }
+
+		/// <summary>
+		/// Create JsonPatch from difference between original and modified
+		/// </summary>
+		/// <param name="originalObject">Original object</param>
+		/// <param name="modifiedObject">Modified object</param>
+		/// <returns>JsonPatch</returns>
+		/// <exception cref="ArgumentException">Only Object and Array type json object allowed</exception>
+        public static JsonPatch CreatePatch(this JsonElement originalObject, JsonElement modifiedObject)
+        {
+            var patch = new List<PatchOperation>();
+            if (originalObject.ValueKind == JsonValueKind.Object)
+                PatchForObject(originalObject, modifiedObject, patch, "/");
+            else if (originalObject.ValueKind == JsonValueKind.Array)
+                PatchForArray(originalObject, modifiedObject, patch, "/");
+            else
+                throw new ArgumentException($"{nameof(originalObject)} should be Object or Array Type.");
+            return new JsonPatch(patch);
+        }
+
+        private static void PatchForObject(JsonElement orig, JsonElement mod, List<PatchOperation> patch, string path)
```

Comment:
I think I'd like to see the path represented as a `JsonPointer` here rather than a string.  Check out `JsonPointer.Create()` and `JsonPointer.Combine()` for managing them.  This would be a lot faster than going through parsing each time.

Reply:
I think I agree with Combine, but what with Create? It needs known type to perform, because it uses expression with exact type. In this case I can only use Combine+Parse

Reply:
There's a non-generic `Create` that you can use as well that just takes segments (looks like I still need to define implicit conversions from ints and strings to segments).

https://github.com/gregsdennis/json-everything/blob/master/JsonPointer/JsonPointer.cs#L198

Reply:
I'm going to update the pointer code to include a fluent builder mechanism.  You'll need to rebase on top of that.  It shouldn't have any conflicts since it's in a different project.